client/src/app/services/event.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from "@angular/common/http";
+import { User } from "../interfaces/user";
+import { environment } from "../../environments/environment";
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class EventService {
+  private url = environment.eventsUrl;
+  private token: string | boolean = false;
+
+  private headers: HttpHeaders = new HttpHeaders();
+  private options: any;
+
+  constructor(private http: HttpClient) {}
+
+  ngOnInit() {
+    this.setHeaders()
+  }
+
+  private setHeaders() {
+    this.headers = new HttpHeaders({
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer ' + this.token
+    });
+    this.options = { headers: this.headers };
+  }
```

Comment:
This can be implemented with [interceptors](https://medium.com/@jaydeepvpatil225/http-interceptors-in-angular-6e9891ae0538)